plugins/inventory/foreman.py

```diff
@@ -238,10 +296,203 @@ def _get_hostvars(self, host, vars_prefix='', omitted_vars=()):
+            for group in ['environment', 'location', 'organization']:
+                val = host.get('%s' % group)
+                if val:
+                    safe_key = to_safe_group_name('%s%s_%s' % (
+                        to_text(self.group_prefix),
+                        group,
+                        to_text(val).lower()
+                    ))
+                    env_lo_org = self.inventory.add_group(safe_key)
+                    self.inventory.add_child(env_lo_org, host_name)
+
+            for group in ['lifecycle_environment', 'content_view']:
+                val = content_facet_attributes.get('%s_name' % group)
+                if val:
+                    safe_key = to_safe_group_name('%s%s_%s' % (
+                        to_text(self.group_prefix),
+                        group,
+                        to_text(val).lower()
+                    ))
+                    le_cv_group = self.inventory.add_group(safe_key)
+                    self.inventory.add_child(le_cv_group, host_name)
+            params = host_params
+
+            if self.want_hostcollections:
+                hostcollections = host.get('host_collections')
+
+                if hostcollections:
+                    # Create Ansible groups for host collections
+                    for hostcollection in hostcollections:
+                        try:
+                            host_collection_group_name = to_safe_group_name('%shostcollection_%s' % (
+                                to_text(self.group_prefix),
+                                to_text(hostcollection).lower()
+                            ))
+                            hostcollection_group = self.inventory.add_group(host_collection_group_name)
+                            self.inventory.add_child(hostcollection_group, host_name)
+                        except ValueError as e:
+                            self.display.warning("Could not create groups for host collections for %s, skipping: %s" % (host_name, to_text(e)))
 
-        for host in self._get_hosts():
+            # set host vars from params
+                if self.get_option('want_params'):
+                    filtered_params = {}
+                    for p in params:
+                        if 'name' in p and 'value' in p:
+                            filtered_params[p['name']] = p['value']
 
+                    if self.get_option('legacy_hostvars'):
+                        self.inventory.set_variable(host_name, 'foreman_params', filtered_params)
+                    else:
+                        for k, v in filtered_params.items():
+                            try:
+                                self.inventory.set_variable(host_name, k, v)
+                            except ValueError as e:
+                                self.display.warning("Could not set hostvar %s to '%s' for the '%s' host, skipping:  %s" %
+                                                     (k, to_native(v), host, to_native(e)))
```

Comment:
I think the code should be indented at the same level as `# set host vars from params`, as otherwise it's only executed when `self.want_hostcollections` is `True`, which seems not what it should do?